lib/virt_autotest/utils.pm

```diff
@@ -567,4 +568,37 @@ EOF
     return;
 }
 
+=head2 check_port_state
+
+Check whether given port is open on remote machine. This subroutine accepts four arguments, dst_machine, dst_port, 
+retries and delay, which are fqdn or ip addr of remote machine, port on remote machine, the number of retries and 
+delay value respectively. Default retries is 1 and default delay is 30 seconds. dst_machine and dst_port have no 
+default value and test will die if the subroutine is called without being passed value to dst_machine or dst_port.
+The subroutine will return 1 if the given port is open on the specified remote machine, otherwise it will return 0.
+
+=cut
+sub check_port_state {
+    my ($dst_machine, $dst_port, $retries, $delay) = @_;
+    $dst_machine //= "";
+    $dst_port    //= "";
+    $retries     //= 1;
+    $delay       //= 30;
+    croak("IP address or FQDN should be given to argument dst_machine or port number should be given to argument dst_port.") if (($dst_machine eq "") or ($dst_port eq ""));
+
+    my $port_state = 0;
+    foreach (1 .. $retries) {
+        save_screenshot;
+        if(IO::Socket::INET->new(PeerAddr => "$dst_machine", PeerPort => "$dst_port")) {
+            save_screenshot;
+            record_info("Port $dst_port is open on machine $dst_machine", "Good luck !");
+            $port_state = 1;
+            last;
+        }
+        save_screenshot;
+        sleep $delay if ($delay > 0);
+    }
+    record_info("Port $dst_port is not open on machine $dst_machine", "Very sorry !") if ($port_state == 0);
```

Comment:
```suggestion
    record_info('Port not open', "$dst_port on machine $dst_machine is not open") if $port_state == 0;
```

the first string to `record_info` is shown in the thumbnail preview and has only very limited space. When clicking on the text box more details can be shown. This is what the second string is for. I recommend to avoid the text `Very sorry !` as I doubt it's helpful to test reviewers (also wrong grammar with the space for the exclamation mark and the exclamation mark itself).

Would be good to see this in a verification run so that we can see it in action

Reply:
I already have some runs but with old text, for example, [this one](http://10.67.19.99/tests/147). I am doing more full verification run with the latest change.

Reply:
[Here](http://10.67.19.99/tests/152) is the latest run with the latest change.